src/Akka.Hosting/Configuration/HoconConfigurationRoot.cs

```diff
@@ -0,0 +1,74 @@
+﻿// -----------------------------------------------------------------------
+//   <copyright file="HoconConfigurationRoot.cs" company="Akka.NET Project">
+//      Copyright (C) 2013-2022 .NET Foundation <https://github.com/akkadotnet/akka.net>
+//   </copyright>
+// -----------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Configuration.EnvironmentVariables;
+using Microsoft.Extensions.Primitives;
+
+namespace Akka.Hosting.Configuration;
+
+internal class HoconConfigurationRoot: IConfigurationRoot
+{
+    private readonly IList<IConfigurationProvider> _providers;
+
+    public HoconConfigurationRoot(IEnumerable<IConfigurationProvider> providers)
+    {
+        _providers = providers.Select(provider => 
+            provider is not EnvironmentVariablesConfigurationProvider envProvider ? 
+                provider : EnvironmentVariablesConfigurationHoconAdapter.Create(envProvider)).ToList();
```

Comment:
We're doing a provider list copy here, replacing/wrapping the official `EnvironmentVariableConfigurationProvider` with our own